compiler/pipes/check-func-calls-and-vararg.cpp

```diff
@@ -87,6 +88,72 @@ VertexAdaptor<op_func_call> CheckFuncCallsAndVarargPass::process_varargs(VertexA
   return new_call;
 }
 
+// when calling f($arg) like f() (without $arg), then maybe, this $arg is auto-filled
+VertexPtr CheckFuncCallsAndVarargPass::maybe_autofill_missing_call_arg(VertexAdaptor<op_func_call> call, [[maybe_unused]] FunctionPtr f_called, VertexAdaptor<op_func_param> param) {
+  if (param->type_hint == nullptr) {
+    return {};
+  }
+
+  // CompileTimeLocation is a built-in KPHP class
+  // in PHP, we declare f(CompileTimeLocation $loc = null) and just call f()
+  // in KPHP, we implicitly replace f() with f(new CompileTimeLocation(__FILE__, __METHOD__, __LINE__))
+  if (const auto *as_instance = param->type_hint->unwrap_optional()->try_as<TypeHintInstance>()) {
+    if (as_instance->resolve()->name == "CompileTimeLocation") {
+      return CheckFuncCallsAndVarargPass::create_CompileTimeLocation_call_arg(call->location);
+    }
+  }
+
+  return {};
+}
+
+// create constructor call: new CompileTimeLocation(__RELATIVE_FILE__, __METHOD__, __LINE__)
+VertexPtr CheckFuncCallsAndVarargPass::create_CompileTimeLocation_call_arg(const Location &call_location) {
+  ClassPtr klass_CompileTimeLocation = G->get_class("CompileTimeLocation");
+  kphp_assert(klass_CompileTimeLocation);
+
+  auto v_file = VertexAdaptor<op_string>::create().set_location(call_location);
+  v_file->str_val = call_location.get_file()->relative_file_name;
+
+  auto v_function = VertexAdaptor<op_string>::create().set_location(call_location);
+  if (current_function->is_lambda()) {  // like tok_method_c in gentree (PHP polyfill also emits the same format)
+    v_function->str_val = "{closure}";
+  } else if (!current_function->is_main_function()) {
+    v_function->str_val = !current_function->modifiers.is_nonmember()
+                          ? current_function->class_id->name + "::" + current_function->local_name()
+                          : current_function->name;
+  }
+
+  auto v_line = VertexAdaptor<op_int_const>::create().set_location(call_location);
+  v_line->str_val = std::to_string(call_location.get_line());
```

Comment:
How about `GenTree::create_int_const(call_location.get_line())`?

Reply:
Done. Forgot about this function